lib/Checks/TrailingWhitespaceCheck.vala

```diff
@@ -25,19 +25,17 @@ public class ValaLint.Checks.TrailingWhitespaceCheck : Check {
             title: _("trailing-whitespace"),
             description:_("Checks for whitespaces at the end of lines")
         );
-    }
 
-    public override void check (Gee.ArrayList<ParseResult? > parse_result, ref Gee.ArrayList<FormatMistake? > mistake_list) {
-        foreach (ParseResult r in parse_result) {
-            if (r.type == ParseType.Default) {
-                add_regex_mistake ("""\h\n""", _("Unexpected whitespace at end of line"), r, ref mistake_list);
-            }
-        }
+        add_regex_check (
+            ParseType.Default,
+            """\h\n""",
+            _("Unexpected whitespace at end of line")
+        );
 
-        // Check for whitespace at last line
-        ParseResult r_last = parse_result.last ();
-        if (r_last.type == ParseType.Default) {
-            add_regex_mistake ("""\h$""", _("Unexpected whitespace at end of last line"), r_last, ref mistake_list);
```

Comment:
We should check the `\h$` regex only in the last element of the parse_result array. In your version, it would also find a whitespace e.g. before a comment.